baremetal/metal3machine_manager_test.go

```diff
@@ -597,103 +651,169 @@ var _ = Describe("Metal3Machine manager", func() {
 					return
 				}
 				Expect(err).NotTo(HaveOccurred())
-				if result != nil {
+				if tc.ExpectedHostName != "" {
 					Expect(result.Name).To(Equal(tc.ExpectedHostName))
 				}
 			},
-			Entry("Pick host2 which lacks ConsumerRef", testCaseChooseHost{
-				Machine:          newMachine("machine1", "", infrastructureRef2),
-				Hosts:            []client.Object{&host2, &host1},
-				M3Machine:        m3mconfig2,
-				ExpectedHostName: host2.Name,
-			}),
```

Comment:
is this test case covered somewhere else that I missed maybe or if not let's put it back?

Reply:
Disregard, it has been covered already on line 715